### PR TITLE
User career refactoring

### PR DESCRIPTION
**Pull request by ace0625**
- Fix UI in User career pages and some code refactoring.

### CODE REVIEW

File: src/components/UserList/Row.js [hunk 1]
```diff
@@ -72,13 +72,11 @@ class Row extends Component {
   }
 
   getProfileImage() {
-    let image;
     if (this.props.dataSource.profile_picture) {
-      image = { uri: this.props.dataSource.profile_picture };
+      return { uri: this.props.dataSource.profile_picture };
       return image;
```
Comment (MechanicKim) [inline src/components/UserList/Row.js]: Double return!

File: src/components/userProfile/CareerRow.js [hunk 2]
```diff
@@ -10,73 +11,129 @@ class CareerRow extends Component {
     super(props);
 
     this.state = {
-      schoolName: '',
-      major: '',
-      companyName: '',
-      position: '',
-      period: '',
-      identifier: 0,
+      schoolName: undefined,
```
Comment (MechanicKim) [inline src/components/userProfile/CareerRow.js:14]: Interesting link about undefined eslint rule.
http://eslint.org/docs/rules/no-undefined
`undefined -> void 0`

File: src/components/userProfile/CareerRow.js [hunk 3]
```diff
@@ -10,73 +11,129 @@ class CareerRow extends Component {
...
+        schoolName: this.getSchoolName(),
+        major: this.getMajor(),
+        graduationYear: this.graduationYear(),
+        loaded: true,
+      });
+    }
+  }
+
+  getCompanyName() {
     let companyName = this.state.companyName;
```
Comment (MechanicKim) [inline src/components/userProfile/CareerRow.js]: You can write this function like this.

```
if (this.props.dataSource.employer) {
  return this.props.dataSource.employer.name;
} else {
  return this.state.companyName;
}
```

or

```
if (this.props.dataSource.employer) {
  return this.props.dataSource.employer.name;
}
 
return this.state.companyName;
```

File: src/components/UserList/Row.js (hunk 1 above)
Comment (ifndefdeadmau5) [inline src/components/UserList/Row.js]: Unreachable return statement

File: src/components/userProfile/CareerRow.js [hunk 4]
```diff
@@ -10,73 +11,129 @@ class CareerRow extends Component {
...
+    }
 
-      if (this.props.dataSource.start_date) {
-        period = this.props.dataSource.start_date + ' - ' + this.props.dataSource.end_date;
-      }
-    } else if (this.props.dataSource.school) {
+    return companyName;
+  }
+
+  getPosition() {
```
Comment (MechanicKim) [inline src/components/userProfile/CareerRow.js:50]: You can write this function like this.
```
if (this.props.dataSource.position) {
  return this.props.dataSource.position.name;
} else {
  return this.state.position;
}
```
or
```
if (this.props.dataSource.position) {
  return this.props.dataSource.position.name;
}
return this.state.position;
```

File: src/components/userProfile/CareerRow.js [hunk 5]
```diff
@@ -10,73 +11,129 @@ class CareerRow extends Component {
...
-      position: position,
-      period: period,
-      identifier: identifier,
-      loaded: true,
-    });
+    return major;
   }
 
-  renderEducation() {
+  graduationYear() {
```
Comment (ifndefdeadmau5) [inline src/components/userProfile/CareerRow.js:82]: How about rename this function to have a verb form? Traditionally we use verb when make a function name

File: src/components/userProfile/CareerRow.js [hunk 6]
```diff
@@ -10,73 +11,111 @@ class CareerRow extends Component {
...
-        <Text style ={styles.period}>{this.state.period}</Text>
+        {this.state.position !== undefined &&
+          <Text style ={styles.name}>{this.state.position}</Text>
+        }
+        {this.state.companyName !== undefined &&
+          <Text style ={styles.name}>{this.state.companyName}</Text>
+        }
+        {this.state.period !== undefined &&
+          <Text style ={styles.period}>{this.state.period}</Text>
+        }
```
Comment (ifndefdeadmau5) [inline src/components/userProfile/CareerRow.js:101]: Interesting code here!
You guys can check some information about this in [here](https://facebook.github.io/react/docs/conditional-rendering.html)

File: src/components/userProfile/CareerRow.js [hunk 7]
```diff
@@ -40,69 +40,51 @@ class CareerRow extends Component {
...
-    let graduationYear = this.state.graduationYear;
-
     if (this.props.dataSource.year && this.props.dataSource.year.name !== '') {
-      graduationYear = 'Class of ' + this.props.dataSource.year.name;
-      return graduationYear;
+      return 'Class of ' + this.props.dataSource.year.name;
     }
 
-    return graduationYear;
+    return this.state.graduationYear;;
```
Comment (ifndefdeadmau5) [inline src/components/userProfile/CareerRow.js:87]: Two semicolon?

File: src/components/userProfile/CareerRow.js [hunk 8]
```diff
@@ -40,69 +40,51 @@ class CareerRow extends Component {
...
-    let schoolName = this.state.schoolName;
-
     if (this.props.dataSource.school) {
-      schoolName = this.props.dataSource.school.name;
-      return schoolName;
+      return this.props.dataSource.school.name;
     }
 
-    return schoolName;
+    return this.state.schoolName;;
```
Comment (ifndefdeadmau5) [inline src/components/userProfile/CareerRow.js:71]: Unnecessary semicolon

**Comment by ifndefdeadmau5** [comment]
LGTM

[Approved with PullApprove](https://pullapprove.com/sv-bootcamp/bridgeme-client/pull-request/117/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=sv-bootcamp/bridgeme-client)

**Pull request merged**